core/src/main/java/com/linecorp/armeria/client/HttpSessionHandler.java

```diff
@@ -105,6 +107,9 @@
 
         this.channelPool = requireNonNull(channelPool, "channelPool");
         this.channel = requireNonNull(channel, "channel");
+        final SocketAddress remoteAddress = channel.remoteAddress();
+        assert remoteAddress != null;
```

Comment:
I think we normally use assert when it's obvious in the same method / class. How about using requireNonNull like the others since it may still have a programming error?

Reply:
I thought it was obvious that the `remoteAddress` is not `null` because this is called right after the connection is established by the `eventLoop` and there's no reading event between that. So the socket is not marked as closed which means that we can always bring the `remoteAddress`.
But I think `requireNonNull` is fine as well, so let me use that. Thanks! :)